Snowflake.Data.Tests/SFPutGetTest.cs

```diff
@@ -47,97 +52,157 @@ public void TestPutCommand(string compressionType)
+                    string filePath = Path.GetTempPath() + Guid.NewGuid().ToString() + ".csv." + compressionType;
+                    // Write row data to temp file
+                    File.WriteAllText(filePath, ROW_DATA);
 
-                // Windows user contains a '~' in the path which causes an error
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-                {
-                    string fileName = filePath.Substring(filePath.LastIndexOf('\\') + 1);
-                    putQuery = $"PUT file://C:\\\\Users\\{Environment.UserName}\\AppData\\Local\\Temp\\{fileName} @{DATABASE_NAME}.{SCHEMA_NAME}.%{TEST_TEMP_TABLE_NAME}";
-                    // getQuery = $"GET @{DATABASE_NAME}.{SCHEMA_NAME}.%{TEST_TEMP_TABLE_NAME} file://{tempDirectory}";
-                }
+                    string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                    Directory.CreateDirectory(tempDirectory);
 
-                using (DbCommand command = conn.CreateCommand())
-                {
-                    // Create temp table
-                    command.CommandText = createTable;
-                    command.ExecuteNonQuery();
-
-                    // Upload file
-                    command.CommandText = putQuery;
-                    DbDataReader reader = command.ExecuteReader();
-                    while (reader.Read())
+                    string putQuery = "";
+                    if (stageType == USER_STAGE)
                     {
-                        // Check file status
-                        Assert.AreEqual(reader.GetString(4), UPLOADED);
-                        // Check source and destination compression type
-                        Assert.AreEqual(reader.GetString(6), compressionType);
-                        Assert.AreEqual(reader.GetString(7), compressionType);
+                        putQuery = $"PUT file://{filePath} @~/";
                     }
-
-                    // Copy into temp table
-                    command.CommandText = copyInto;
-                    command.ExecuteNonQuery();
-
-                    // Check contents are correct
-                    command.CommandText = $"SELECT * FROM {TEST_TEMP_TABLE_NAME}";
-                    reader = command.ExecuteReader();
-                    while (reader.Read())
+                    else if (stageType == TABLE_STAGE)
                     {
-                        Assert.AreEqual(reader.GetString(0), COL1_DATA);
-                        Assert.AreEqual(reader.GetString(1), COL2_DATA);
-                        Assert.AreEqual(reader.GetString(2), COL3_DATA);
+                        putQuery = $"PUT file://{filePath} @{DATABASE_NAME}.{SCHEMA_NAME}.%{TEST_TEMP_TABLE_NAME}";
                     }
+                    else if (stageType == NAMED_STAGE)
+                    {
+                        putQuery = $"PUT file://{filePath} @{DATABASE_NAME}.{SCHEMA_NAME}.{TEST_TEMP_STAGE_NAME}";
```

Comment:
For PUT/GET, should test if GET can work with both AUTO_COMPRESS = TRUE | FALSE following https://docs.snowflake.com/en/sql-reference/sql/put.html